### PR TITLE
Update the Docker configuration

### LINKED ISSUE #276: Synchronize configuration before the example and the Docker image
Currently the Docker image uses a separate ``production.yaml`` with sane default production values.
Unfortunately, that means that any new configuration setting needs to be mirrored in that file; this causes updates on ``develop`` to break the image because configuration cannot be found when loading the file.

I suggest we use the default configuration file in the image while customizing it a bit for production during build.

### PR DESCRIPTION
**Pull request by kaiyou**
This pull requests updates the Docker production files, including:

- the build files for alpine and Debian, using the default configuration file as a base which fixes #276
- the compose file, fixing some typos

This also reverts back to copying files from the local repository. I have a strong opinion about being able to build any version by checking out that version and running Docker build, which cloning the repo inside the Dockefile does not allow. I believe this is just as useful when using Docker Hub to build tag and branch versions.

### CODE REVIEW

File: support/docker/production/docker-compose.yml [hunk 1]
```diff
@@ -26,7 +23,7 @@ services:
     # remove them if you are using another reverse proxy.
     labels:
       traefik.enable: "true"
-      traefik.frontend.rule: "Host:${PEERTUBE_HOSTNAME}"
+      traefik.frontend.rule: "Host:my.domain.tld"
```
[the remark this answers is not in the document]
Comment (rigelk) [reply support/docker/production/docker-compose.yml]: Fait enough.

File: support/docker/production/docker-compose.yml [hunk 2]
```diff
@@ -1,14 +1,11 @@
...
+version: "3.1"
 
 services:
 
   peertube:
-    build:
-      context: .
-      dockerfile: ./Dockerfile.stretch
-    image: chocobozzz/peertube:production-stretch
+    image: chocobozzz/peertube:production
```
Comment (Chocobozzz) [inline support/docker/production/docker-compose.yml]: Please keep the `stretch`

**Comment by Chocobozzz** [comment]
Implemented in another commit: https://github.com/Chocobozzz/PeerTube/commit/4aeb67a569b30a233dcff2eaf9c564d62ccd6497#diff-1a1a8c4751f9ceb8309879af20a2fa50R54

**Comment by kaiyou** [comment]
This is not the only part the pull request was meant to address.

Are you opposed to build from the repo content instead of downloading a fresh source?

**Comment by Chocobozzz** [comment]
We build from the repo content now (see the line I linked)

**Comment by kaiyou** [comment]
OK I missed that part, perfect then.

**Pull request closed without merging**